lib/priority_queue.dart

```diff
@@ -13,6 +13,16 @@ import "dart:collection" show SplayTreeSet;
  */
 abstract class PriorityQueue<E> {
   /**
+   * Creates an empty [PriorityQueue].
+   *
+   * The created [PriorityQueue] is a plain [HeapPriorityQueue]. As such,
+   * [Comparable.compare] is used to compare the priority of elements.
+   *
+   * The queue is equivalent to one created by `new HeapPriorityQueue<E>()`.
+   */
+  factory PriorityQueue() = HeapPriorityQueue<E>;
```

Comment:
I used the same pattern as `Set()` that forward to `LinkedHashSet` (https://github.com/dart-lang/sdk/blob/master/sdk/lib/core/set.dart#L45 and https://github.com/dart-lang/sdk/blob/master/sdk/lib/collection/linked_hash_set.dart#L77). Let me know what you prefer.